### PR TITLE
Java 11 support

### PR DESCRIPTION
**Pull request by hwbllmnn**
Based on #294. Currently the tests run again with Java 11. Javadoc generation has `failOnError` disabled because of numerous linting errors. We should test this in an actual project with Java 11 before merging (and think about whether we want to maintain a 'stable' 1.x or 1.0.x branch).

Java 9/10 support has been disabled in travis for now since Java 11 is the new LTS version anyway.

@terrestris/devs any thoughts?

### CODE REVIEW

**Comment by buehner** [comment]
nice work! :+1: for merging this after testing it with a java 9 project

File: src/pom.xml [hunk 1]
```diff
@@ -143,11 +143,10 @@
         <slf4j.version>1.7.25</slf4j.version>
         <jackson.version>2.9.3</jackson.version>
         <opencsv.version>4.1</opencsv.version>
-        <ehcache.version>3.4.0</ehcache.version>
+        <ehcache.version>3.5.0</ehcache.version>
```
Comment (ahennr) [inline src/pom.xml]: You use `JCacheRegionFactory` in db context for `"hibernate.cache.region.factory_class`. Is ecache needed anymore?

Comment (hwbllmnn) [reply src/pom.xml]: Yes, ehcache brings a JCache implementation.

**Comment by buehner** [comment]
Thx! Is it possible to resolve the conflicts?

**Comment by hwbllmnn** [comment]
Done. I've also made a few tweaks to the caching annotations & configuration in order to remove lots of warnings.

**Comment by buehner** [comment]
OK. I got this running succesfully (in the context of a project) and will merge now

**Pull request merged**